packages/rich-text/src/helpers/editor.ts

```diff
@@ -124,25 +129,31 @@ export function unwrapLink(editor) {
     match: (node) =>
       !Editor.isEditor(node) &&
       Element.isElement(node) &&
-      (node as CustomElement).type === INLINES.HYPERLINK, // TODO: Support Entry and Asset links
+      LINK_TYPES.includes((node as CustomElement).type), // TODO: Support Entry and Asset links
   });
 }
 
-export function wrapLink(editor, text, url) {
+export function wrapLink(editor, { text, url, target, type }: InsertLinkOptions) {
   if (isLinkActive(editor)) {
     unwrapLink(editor);
   }
 
   const { selection } = editor;
   const isCollapsed = selection && Range.isCollapsed(selection);
   const link = {
-    type: INLINES.HYPERLINK, // TODO: Support Entry and Asset links
-    data: {
-      uri: url,
-    },
+    type,
+    data: {},
     children: isCollapsed ? [{ text }] : [],
   };
 
+  if (url) {
+    link.data = { url };
+  }
+
+  if (target) {
+    link.data = { target };
+  }
```

Comment:
Would we face any issues with a scenario where `url` and `target` are both present?

Reply:
I don't think we can both at the same type. Nevertheless, `type` will dictate what to render, and the render will either use `target` or `uri`.